### PR TITLE
feat: Implement multiple AllowMixedWeaponType options

### PR DESCRIPTION
**Pull request by ZhengPeiRu21**
Correctly do allow Main-hand Weapons and Off-hand Weapons to only be allowed to transmog to weapons of the same inventory type.

### CODE REVIEW

File: src/Transmogrification.cpp [hunk 1]
```diff
@@ -470,13 +470,62 @@ bool Transmogrification::CanTransmogrifyItemWithItem(Player* player, ItemTemplat
...
                 return false;
-            if (source->Class == ITEM_CLASS_WEAPON && !AllowMixedWeaponTypes)
-                return false;
+            if (source->Class == ITEM_CLASS_WEAPON)
+            {
+                if (AllowMixedWeaponTypes == MIXED_WEAPONS_STRICT)
+                {
+                    return false;
+                }
+                else if (AllowMixedWeaponTypes == MIXED_WEAPONS_MODERN)
```
Comment (Helias) [inline src/Transmogrification.cpp:479]: if you have an if condition with return, it is not necessary the `else`
```suggestion
                if (AllowMixedWeaponTypes == MIXED_WEAPONS_MODERN)
```

File: src/Transmogrification.cpp [hunk 2]
```diff
@@ -801,9 +850,10 @@ void Transmogrification::LoadConfig(bool reload)
     AllowTradeable = sConfigMgr->GetOption<bool>("Transmogrification.AllowTradeable", false);
 
     AllowMixedArmorTypes = sConfigMgr->GetOption<bool>("Transmogrification.AllowMixedArmorTypes", false);
-    AllowMixedWeaponTypes = sConfigMgr->GetOption<bool>("Transmogrification.AllowMixedWeaponTypes", false);
     AllowFishingPoles = sConfigMgr->GetOption<bool>("Transmogrification.AllowFishingPoles", false);
 
+    AllowMixedWeaponTypes = sConfigMgr->GetOption<uint8>("Transmogrification.AllowMixedWeaponTypes", 0);
```
Comment (Helias) [inline src/Transmogrification.cpp:855]: What about putting `MIXED_WEAPONS_STRICT` as default value? it's always 0 but more readable
```suggestion
    AllowMixedWeaponTypes = sConfigMgr->GetOption<uint8>("Transmogrification.AllowMixedWeaponTypes", MIXED_WEAPONS_STRICT);
```

File: src/Transmogrification.h [hunk 3]
```diff
@@ -31,6 +31,13 @@ enum TransmogSettings
     SETTING_RETROACTIVE_CHECK = 1
 };
 
+enum MixedWeaponSettings
+{
+    MIXED_WEAPONS_STRICT = 0,
+    MIXED_WEAPONS_MODERN = 1,
+    MIXED_WEAPONS_LOOSE  = 2
```
Comment (Helias) [inline src/Transmogrification.h:38]: the enumeration starts always from 0, so you can also remove the assignment IIRC
```suggestion
    MIXED_WEAPONS_STRICT,
    MIXED_WEAPONS_MODERN,
    MIXED_WEAPONS_LOOSE
```

**Comment by Helias** [comment]
thanks for this PR and for the job that you did to transmog so far, it's very nice 🚀 

I left some comments to improve the code quality and readability but nothing really important, these comments are not blocker for this PR, if you want to implement them, it's ok, otherwise it's also ok 👍

**Comment by Gultask** [comment]
Also it might be good to rename the PR to
feat: Implement multiple AllowMixedWeaponType options
:D

File: src/Transmogrification.h (hunk 3 above)
Comment (ZhengPeiRu21) [reply src/Transmogrification.h:38]: I think having the values here may be helpful since they will show which setting corresponds to which config value.

**Comment by ZhengPeiRu21** [comment]
> thanks for this PR and for the job that you did to transmog so far, it's very nice 🚀
> 
> I left some comments to improve the code quality and readability but nothing really important, these comments are not blocker for this PR, if you want to implement them, it's ok, otherwise it's also ok 👍

Thank you for your helpful reviews! I have committed your suggestions.

**Comment by Gultask** [comment]
I don't think I had any problems with it. All the restrictions worked perfectly from what I've seen and the MH/OF restrictions as well.
Switching the config files also worked the same as it did before for both settings.

Switching the config and having restricted weapon types transmogged didn't give any warnings or errors as well, which is fine.

So it looks good from my end 👍

**Pull request merged**